app/gui2/shared/yjsModel.ts

```diff
@@ -99,16 +93,26 @@ export class DistributedProject {
   }
 }
 
+export interface NodeMetadata {
+  x: number
+  y: number
+  vis: VisualizationMetadata | null
+}
+
+export type ModuleSetting = string
+
 export class ModuleDoc {
   ydoc: Y.Doc
   contents: Y.Text
   idMap: Y.Map<Uint8Array>
   metadata: Y.Map<NodeMetadata>
+  settings: Y.Map<ModuleSetting>
   constructor(ydoc: Y.Doc) {
     this.ydoc = ydoc
     this.contents = ydoc.getText('contents')
     this.idMap = ydoc.getMap('idMap')
     this.metadata = ydoc.getMap('metadata')
+    this.settings = ydoc.getMap('settings')
```

Comment:
I believe that this should be tied to a whole project doc, not each module. Otherwise it would be possible to set different execution modes when viewing different files, which sounds incorrect. The modules can call each other after all.